### PR TITLE
Adding a chRoot for provided zookeepers.

### PR DESCRIPTION
**Pull request by HoustonPutman**
*Issue number of the reported bug or feature request: #61 *

**Describe your changes**
Added the possiblity to use chRoot for a provided zookeeper.

The zookeeperConnectionString for provided zookeepers was also changed to use the service name instead of the ip address. This should lower the number of restarts caused by service IP changes, if there were any in the first place.

**Testing performed**
Added tests around this functionality. Also added tests for other environment variables.

### CODE REVIEW

**Comment by HoustonPutman** [comment]
@sepulworld want to give this a shot?

I'm not running a solr w/ a provided zookeeper right now, but I could spin one up with that if you don't have the time.

**Comment by sepulworld** [comment]
Ill test it out! ty!

File: README.md [hunk 1]
```diff
@@ -390,10 +390,14 @@ kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/mast
...
 ## Version Compatability & Changes
 
+#### v0.2.1
+- The zkConnectionString used for provided zookeepers changed from using the string provided in the `ZkCluster.Status`, which used an IP, to using the service name. This will cause a rolling restart of your solrs using the provided zookeeper option, but there will be no data loss.
+
 #### v0.2.0
 - Uses `gomod` instead of `dep`
 - `SolrCloud.zookeeperRef.provided.zookeeper.persistentVolumeClaimSpec` has been deprecated in favor of the `SolrCloud.zookeeperRef.provided.zookeeper.persistence` option.  
 This option is backwards compatible, but will be removed in a future version.
+- An upgrade to the ZKOperator version `0.2.4` is required.
```
Comment (sepulworld) [inline README.md]: Good call, perhaps update https://github.com/bloomberg/solr-operator/blob/master/example/dependencies/operators.yaml#L50 to 0.2.4 as well?

Comment (HoustonPutman) [reply README.md]: Good call for sure.

**Comment by HoustonPutman** [comment]
Ok I think this is ready to go. I have tested it on my end with new chRoots for both external ZKs and provided ZKs.

**Comment by sepulworld** [comment]
> Ok I think this is ready to go. I have tested it on my end with new chRoots for both external ZKs and provided ZKs.

This is great! Thank you!

**Comment by sepulworld** [comment]
Works well!

```
  zookeeperRef:
    provided:
      chroot: "example"
```

Screen Shot 2020-01-09 at 3 32 14 PM

**Pull request merged**